pkg/route/api/v1/types.go

```diff
@@ -48,6 +48,13 @@ type RouteSpec struct {
 
 	// TLS provides the ability to configure certificates and termination for the route
 	TLS *TLSConfig `json:"tls,omitempty" description:"provides the ability to configure certificates and termination for the route"`
+
+	// RouterShard has information of a routing shard and is used to
+	// generate host names
+	Shard *RouterShard `json:"shard,omitempty" description:"provides shard information for the route"`
+
+	// If specfied and true then DNSSuffix is appended to display in UI when host is specified for a route
+	AppendDNSSuffix bool `json:"appendDNSSuffix,omitempty" description:"provides information whether to append DNSSuffix to display in UI when host is specified"`
```

Comment:
Just to clarify one thing, specifying custom route host works properly as far as router functionality is concerned, just need to add  global suffix to it while accessing it. Means UI displays spec.host as etcd-test-discovery in current version of openshift. But the endpoint can be accessed via http://etcd-test-discovery.router.default.svc.cluster.local/. Just need to add the global suffix at the end of custom route host while accessing the endpoint.